### ISSUE TITLE
Upgraded to 0.4.0 and cannot read files in ZipFS

### ISSUE THREAD

**Issue by atty303** [zen-fs/zip#4]
```ts
    const rootZip = await fetch(`https://pob-web-asset.atty303.ninja/versions/v2.42.0/root.zip`);
    await zenfs.configure({
      mounts: {
        "/root": {
          backend: Zip,
          data: await rootZip.arrayBuffer(),
          name: "root.zip",
        },
      },
    });

    console.log(await zenfs.promises.exists("/root/.image.tsv"));
```

This causes the following error.

```
RangeError: Offset is outside the bounds of the DataView
    at DataView.prototype.getUint32 (<anonymous>)
    at deserialize (utilium.js?v=defcc128:413:29)
    at new FileEntry (zip.js:212:9)
    at new ZipFS (ZipFS.js:84:24)
    at Object.create (ZipFS.js:175:16)
    at resolveMountConfig (chunk-N5MENFCO.js?v=defcc128:13743:32)
    at async Module.configure (chunk-N5MENFCO.js?v=defcc128:13767:28)
    at async DriverWorker.start (worker.ts?worker_file&type=module:79:9)
```

This exception itself stopped appearing after a simple (but wrong) change.

```diff
-			const cd = new FileEntry(this.data, this.data.slice(ptr, sizeof_FileEntry));
+			const cd = new FileEntry(this.data, this.data.slice(ptr));
```

Another exception will be made.

```
 Error: No such file or directory
    at _ErrnoError.With (http:/root/rootlocalhost:5173/rootnode_modules/root.vite/rootdeps/rootchunk-H4GO7QOY.js?v=defcc128:163:12)
    at ZipFS.statSync (http:/root/rootlocalhost:5173/root@fs/rootC:/rootUsers/rootatty/rootsrc/rootgithub.com/rootatty303/rootzenfs-zip/rootdist/rootZipFS.js:111:30)
    at ZipFS.stat (http:/root/rootlocalhost:5173/rootnode_modules/root.vite/rootdeps/root@zenfs_core_filesystem__js.js?v=defcc128:473:19)
    at realpath (http:/root/rootlocalhost:5173/rootnode_modules/root.vite/rootdeps/rootchunk-N5MENFCO.js?v=defcc128:13349:28)
    at realpath (http:/root/rootlocalhost:5173/rootnode_modules/root.vite/rootdeps/rootchunk-N5MENFCO.js?v=defcc128:13346:47)
    at Object.exists (http:/root/rootlocalhost:5173/rootnode_modules/root.vite/rootdeps/rootchunk-N5MENFCO.js?v=defcc128:13098:55)
    at DriverWorker.start (http:/root/rootlocalhost:5173/root@fs/rootC:/rootUsers/rootatty/rootsrc/rootgithub.com/rootatty303/rootpob-web/rootpackages/rootdriver/rootsrc/rootjs/rootworker.ts?worker_file&type=module:88:42)
```

The same exception occurs with `zenfs.promises.exists("/root")`, so it seems to be caused by the absence of "/" in the ZipFS entries.

**Comment by dr-vortex** [comment]
I'm sorry—[v0.4.1](https://github.com/zen-fs/zip/releases/tag/v0.4.1) should fix this. Perhaps some tests are needed.

**Comment by dr-vortex** [comment]
@atty303 Does v0.4.1 fix this?

**Comment by dr-vortex** [comment]
@atty303,


```diff
-			const cd = new FileEntry(this.data, this.data.slice(ptr));
+			const cd = new FileEntry(this.data, this.data.slice(ptr, sizeof_FileEntry));
```
In v0.4.0, I made this change to slice the entry at the correct size since the previous behavior would slice from `ptr` until the end of the file.

This, for some reason, causes an error to be thrown [here](https://github.com/dr-vortex/utilium/blob/main/src/struct.ts#L189).

I'm investigating this issue more.

**Comment by dr-vortex** [comment]
@atty303 I've found the root cause of the issue, and corrected it: [utilium 6cf8815](https://github.com/dr-vortex/utilium/commit/6cf88154c4aed33948dae1ef6419319c73161287)

**Comment by dr-vortex** [comment]
This has been updated and is now tested.
